crates/ntx-builder/src/builder/coordinator.rs

```diff
@@ -0,0 +1,205 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use anyhow::Context;
+use miden_node_proto::domain::account::NetworkAccountPrefix;
+use miden_node_proto::domain::mempool::MempoolEvent;
+use tokio::sync::mpsc::error::SendError;
+use tokio::sync::{Semaphore, mpsc};
+use tokio::task::JoinSet;
+use tokio_util::sync::CancellationToken;
+
+use crate::actor::{AccountActor, AccountActorConfig, AccountOrigin, ActorShutdownReason};
+
+// ACTOR HANDLE
+// ================================================================================================
+
+/// Handle to account actors that are spawned by the coordinator.
+#[derive(Clone)]
+struct ActorHandle {
+    event_tx: mpsc::Sender<MempoolEvent>, // TODO: consider Arc<MempoolEvent>.
+    cancel_token: CancellationToken,
+}
+
+impl ActorHandle {
+    fn new(event_tx: mpsc::Sender<MempoolEvent>, cancel_token: CancellationToken) -> Self {
+        Self { event_tx, cancel_token }
+    }
+}
+
+// COORDINATOR
+// ================================================================================================
+
+/// Coordinator for managing [`AccountActor`] instances, tasks, and associated communication.
+///
+/// The `Coordinator` is the central orchestrator of the network transaction builder system.
+/// It manages the lifecycle of account actors. Each actor is responsible for handling transactions
+/// for a specific network account prefix. The coordinator provides the following core
+/// functionality:
+///
+/// ## Actor Management
+/// - Spawns new [`AccountActor`] instances for network accounts as needed.
+/// - Maintains a registry of active actors with their communication channels.
+/// - Gracefully handles actor shutdown and cleanup when actors complete or fail.
+/// - Monitors actor tasks through a join set to detect completion or errors.
+///
+/// ## Event Broadcasting
+/// - Distributes mempool events to all relevant account actors.
```

Comment:
My understanding is that currently we distribute events to all actors (i.e., not all **relevant** actors).